meta-oe/recipes-devtools/perfetto/perfetto.bb

```diff
@@ -0,0 +1,116 @@
+LICENSE = "Apache-2.0 & BSD-3-Clause & MIT & ZLib"
+
+LIC_FILES_CHKSUM = "file://LICENSE;md5=f87516e0b698007e9e75a1fe1012b390 \
+                    file://buildtools/libcxx/LICENSE.TXT;md5=55d89dd7eec8d3b4204b680e27da3953 \
+                    file://buildtools/libcxx/utils/google-benchmark/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
+                    file://buildtools/libcxxabi/LICENSE.TXT;md5=7b9334635b542c56868400a46b272b1e \
+                    file://buildtools/libunwind/LICENSE.TXT;md5=f66970035d12f196030658b11725e1a1 \
+                    file://buildtools/protobuf/LICENSE;md5=37b5762e07f0af8c74ce80a8bda4266b \
+                    file://buildtools/zlib/LICENSE;md5=f09575dbfb09420642318b413159496f \
+                    file://debian/copyright;md5=4e08364c82141f181de69d0a2b89d612 \
+                    file://python/LICENSE;md5=c602a632c34ade9c78a976734077bce7"
+
+SRC_URI = "git://github.com/google/perfetto.git;protocol=https;name=perfetto \
+           git://github.com/protocolbuffers/protobuf.git;branch=3.9.x;protocol=https;destsuffix=git/buildtools/protobuf;name=protobuf \
+           git://chromium.googlesource.com/external/github.com/llvm/llvm-project/libcxx.git;protocol=https;destsuffix=git/buildtools/libcxx;name=libcxx \
+           git://chromium.googlesource.com/external/github.com/llvm/llvm-project/libcxxabi.git;protocol=https;destsuffix=git/buildtools/libcxxabi;name=libcxxabi \
+           git://chromium.googlesource.com/external/github.com/llvm/llvm-project/libunwind.git;protocol=https;destsuffix=git/buildtools/libunwind;name=libunwind \
+           git://android.googlesource.com/platform/external/zlib.git;branch=master;protocol=https;destsuffix=git/buildtools/zlib;name=zlib \
+           https://storage.googleapis.com/perfetto/gn-linux64-1968-0725d782;subdir=git/buildtools/;name=gn \
+           file://0001-Remove-check_build_deps-build-steps.patch "
+
+SRCREV_perfetto = "5bd3f582c075d0d026c5fe0b5e291d34dee0d976"
+SRCREV_protobuf = "6a59a2ad1f61d9696092f79b6d74368b4d7970a3"
+SRCREV_libcxx = "d9040c75cfea5928c804ab7c235fed06a63f743a"
+SRCREV_libcxxabi = "196ba1aaa8ac285d94f4ea8d9836390a45360533"
+SRCREV_libunwind = "d999d54f4bca789543a2eb6c995af2d9b5a1f3ed"
+SRCREV_zlib = "5c85a2da4c13eda07f69d81a1579a5afddd35f59"
+SRC_URI[gn.sha256sum] = "f706aaa0676e3e22f5fc9ca482295d7caee8535d1869f99efa2358177b64f5cd"
+
+DEPENDS += " ninja-native"
+
+B = "${WORKDIR}/build"
```

Comment:
this is redundant

Reply:
Sorry I didn't check this section very well, I did a more detailed test for all possible combinations. It seems I need to set both `S` and `B`:

1. Not setting `S` or `B`, without `devtool override -n perfetto /tmp/perfetto`:
`B="/opt/build/tmp/work/armv7a-openbmc-linux-gnueabi/perfetto/1.0-r0/perfetto-1.0"`
`S="/opt/build/tmp/work/armv7a-openbmc-linux-gnueabi/perfetto/1.0-r0/perfetto-1.0"`
--> "patch" task will fail, because source codes are still checked out to ${WORKDIR}/git

2. Not setting S or B, with override
`B="/tmp/perfetto"`
`S="/tmp/perfetto"`
--> "configure" task will fail (because Perfetto doesn't allow the build folder to be the same as the source folder)

3. Only specifying `S = "${WORKDIR}/git"`, without override
`B="/opt/build/tmp/work/armv7a-openbmc-linux-gnueabi/perfetto/1.0-r0/git"`
`S="/opt/build/tmp/work/armv7a-openbmc-linux-gnueabi/perfetto/1.0-r0/git"`
--> "configure" task will fail (same as above)

4. Only specifying S, with override
`B="/tmp/perfetto"`
`S="/tmp/perfetto"`
--> "configure" task will fail (same as above)

5. Only specifying `B = "${WORKDIR}/build"`, without override
`B="/opt/build/tmp/work/armv7a-openbmc-linux-gnueabi/perfetto/1.0-r0/build"`
`S="/opt/build/tmp/work/armv7a-openbmc-linux-gnueabi/perfetto/1.0-r0/perfetto-1.0"`
--> "patch" task will fail, because source codes are still checked out to ${WORKDIR}/git

6. Only specifying B, with override
`B="/tmp/perfetto"`
`S="/tmp/perfetto"`
--> "configure" task will fail

7. Specifying both S and B, no override
`B="/opt/build/tmp/work/armv7a-openbmc-linux-gnueabi/perfetto/1.0-r0/build"`
`S="/opt/build/tmp/work/armv7a-openbmc-linux-gnueabi/perfetto/1.0-r0/git"`
--> works

8. Specifying both S and B, with override
`B="/opt/build/tmp/work/armv7a-openbmc-linux-gnueabi/perfetto/1.0-r0/perfetto-1.0/"`
`S="/tmp/perfetto"`
--> works